### PR TITLE
FPO-145: Adds encryption wrapper class for storing api secrets

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/FPO-145

### What?

I have added/removed/altered:

- [x] Added encryption wrapper for encrypting secrets
- [x] Added specs to cover the behaviour of this wrapper

### Why?

I am doing this because:

- This is to avoid storing plain text values in dynamodb

### CODE REVIEW

File: spec/utils/customerApiKeyEncryptionSpec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { CustomerApiKeyEncryption } from '../../src/utils/customerApiKeyEncryption'
+
+describe('CustomerApiKeyEncription', () => {
+  describe('encrypt', () => {
+    it('should return encrypted key', async () => {
```
Comment (alexdesi) [inline spec/utils/customerApiKeyEncryptionSpec.ts:5]: let's put it in present tense: it ("returns encrypted key", ...

File: spec/utils/customerApiKeyEncryptionSpec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      const encryption = new CustomerApiKeyEncryption()
+
+      const encrypted = await encryption.encrypt('secret')
+
+      expect(encrypted).toMatch(/^[A-Za-z0-9+/]+={0,2}:[A-Za-z0-9+/]+={0,2}$/)
+    })
+  })
+
+  describe('decrypt', () => {
+    it('should return decrypted key', async () => {
```
Comment (alexdesi) [inline spec/utils/customerApiKeyEncryptionSpec.ts:15]: same as before

File: src/utils/customerApiKeyEncryption.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { Buffer } from 'buffer'
+import dotenv from 'dotenv'
+
+dotenv.config({ path: '.env.test' })
+
+export class CustomerApiKeyEncryption {
+  async loadKey(): Promise<crypto.webcrypto.CryptoKey> {
+    let base64Key: string
+
+    if (process.env.ENCRYPTION_KEY === undefined) {
```
Comment (alexdesi) [inline src/utils/customerApiKeyEncryption.ts:11]: we can use:
`if  (process.env.ENCRYPTION_KEY) {`
it is simpler and also
it checks for
not null
and not empty space ""

Comment (willfish) [reply src/utils/customerApiKeyEncryption.ts:11]: I think our linters complain about this

Comment (willfish) [reply src/utils/customerApiKeyEncryption.ts:11]: 2024-03-25-090847_1649x449_scrot

Comment (willfish) [reply src/utils/customerApiKeyEncryption.ts:11]: Yeah, it looks like an unset env key will always be `undefined` or a string

Comment (alexdesi) [inline src/utils/customerApiKeyEncryption.ts:11]: I see ... this weekend I've read about   "non-null assertion operator" 

We could use it and remove the if condition:
`base64Key = process.env.ENCRYPTION_KEY!`

Comment (willfish) [reply src/utils/customerApiKeyEncryption.ts:17]: Typescript doesn't like that either

**Review by alexdesi — APPROVED** [review]
:white_check_mark:

**Pull request merged**